BaseStation/MapTileSetDownloader/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Net;
+using System.IO;
+using System.Collections.Generic;
+
+namespace TileSetDownloader
+{
+    class MainClass
+    {
+        public static void Main(string[] args)
+        {
+            MapGenerator.Configuration config = new MapGenerator.Configuration();
+
+            config.setImageSize(3000, 3000);
+            config.setImageScale(2);
+            config.setZoomLevel(19);
+            config.setLocation(38.4064262, -110.7941097);
+            config.setSatelliteImage();
+
+            MapGenerator.Fetch(config);
+        }
+    }
+
+    class MapGenerator
+    {
+        public class Configuration
+        {
+            Dictionary<String, String> parameters = new Dictionary<String, String>();
+
+            public void setImageSize(int width, int height)
+            {
+                parameters.Add("size", width + "x" + height);
+            }
+
+            public void setImageScale(int scale)
+            {
+                parameters.Add("scale", scale.ToString());
+            }
+
+            public void setLocation(double longitude, double latitude)
+            {
+                parameters.Add("center", longitude + "," + latitude);
+            }
+
+            public void setZoomLevel(int zoom)
+            {
+                parameters.Add("zoom", zoom.ToString());
+            }
+
+            public void setSatelliteImage()
+            {
+                parameters.Add("maptype", "satellite");
+            }
+
+            public String toString()
+            {
+                String result = "";
+                foreach (String key in parameters.Keys)
+                {
+                    result += key + "=" + parameters[key] + "&";
+                }
+                return result;
+            }
+        }
+
+        public static void Fetch(MapGenerator.Configuration config)
```

Comment:
Consider adding parameters for folder name and file name. (excluding the file extension if that's important for working with the Google API.